include/glow/Backends/TraceEvents.h

```diff
@@ -161,6 +166,22 @@ class TraceContext {
   void
   logTraceEvent(llvm::StringRef name, llvm::StringRef type, uint64_t timestamp,
                 std::map<std::string, std::string> additionalAttributes = {});
+
+  /// Sets the human readable \p name for thread \tid.
+  void setThreadName(int tid, llvm::StringRef name);
+  /// \returns the list of human readable thread names.
+  std::map<int, std::string> &getThreadNames() { return threadNames_; }
```

Comment:
why do we expose non const reference to the caller?

Reply:
because they may want to modify it

Reply:
have not seen it modified